### PR TITLE
DGS-104 Functionality added to remove keys without values on serialization

### PR DESCRIPTION
**Pull request by arturas-bes**

### CODE REVIEW

File: src/Api/DTO/Request/ParcelShopSearchRequest.php [hunk 1]
```diff
@@ -209,6 +210,6 @@ public function jsonSerialize()
             $request['id'] =  $this->parcelShopId;
         }
 
-        return $request;
+        return ArrayUtility::removeKeysWithEmptyValues($request);
```
Comment (tomjas1997) [inline src/Api/DTO/Request/ParcelShopSearchRequest.php:208]: Could improve some of the code above as company and parcelShopId can be null now and they will be cleaned off, there's no need for if cases

File: tests/Unit/Utilities/ArrayUtilityTest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    const EXPECTED_RESULT = [
+        'string' => 'value',
+        'integer' => 10,
+        'true' => true,
+    ];
+
+
+
+    public function test_if_array_keys_removed_correctly()
```
Comment (tomjas1997) [inline tests/Unit/Utilities/ArrayUtilityTest.php]: DataProviders are used for these kind of tests :P

File: src/Api/DTO/Request/ParcelShopSearchRequest.php (hunk 1 above)
Comment (arturas-bes) [reply src/Api/DTO/Request/ParcelShopSearchRequest.php:208]: Good point

**Review by tomjas1997 — APPROVED** [review]
Seems OK

File: tests/Unit/Utilities/ArrayUtilityTest.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+use Invertus\dpdBalticsApi\Utilities\ArrayUtility;
+use PHPUnit\Framework\TestCase;
+
+class ArrayUtilityTest extends TestCase
+{
+    /**
+     * @dataProvider getTestableArray
+     */
+    public function test_if_array_keys_removed_correctly($testableArray, $expectedResult)
```
Comment (margud) [inline tests/Unit/Utilities/ArrayUtilityTest.php:11]: hmm why is name with _ and not upper cases?

**Review by margud — APPROVED** [review]

**Pull request merged**